### PR TITLE
Add Joint DOM API to access joint sensors

### PR DESCRIPTION
**Pull request by azeey**
# 🎉 New feature

## Summary
This adds  `sdf::Joint::SensorCount`, `sdf::Joint::SensorByIndex`, `sdf::Joint::SensorNameExists`, `sdf::Joint::SensorByName`.

## Test it
Run tests

## Checklist
- [ ] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] Code check passed (In source directory, run `sh tools/code_check.sh`)
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review
[another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+)
to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: test/integration/joint_dom.cc [hunk 1]
```diff
@@ -857,3 +859,43 @@ TEST(DOMJoint, LoadJointNestedParentChild)
     EXPECT_EQ(Pose(0, -1, 1, IGN_PI_2, 0, 0), pose);
   }
 }
+
+
+//////////////////////////////////////////////////
+TEST(DOMJoint, Sensors)
+{
+  const std::string testFile =
+      sdf::testing::TestFile("sdf", "joint_sensors.sdf");
```
Comment (scpeters) [inline test/integration/joint_dom.cc:868]: I think this file is missing

Comment (azeey) [reply test/integration/joint_dom.cc:868]: I always forget to add the sdf file 🤦🏾‍♂️ . Added in https://github.com/osrf/sdformat/pull/517/commits/8d50882e74804f3dc899dc2892ff90b63475446b

File: src/Joint.cc [hunk 2]
```diff
@@ -405,3 +413,43 @@ sdf::ElementPtr Joint::Element() const
...
+bool Joint::SensorNameExists(const std::string &_name) const
+{
+  for (auto const &s : this->dataPtr->sensors)
+  {
+    if (s.Name() == _name)
+    {
+      return true;
+    }
+  }
+  return false;
```
Comment (jennuine) [inline src/Joint.cc:441]: Could you use `return nullptr != this->SensorByName(_name);` instead?

**Review by jennuine — APPROVED** [review]
LGTM! One minor suggestion

File: src/Joint.cc (hunk 2 above)
Comment (jennuine) [reply src/Joint.cc]: cf387a4f63b5773b5958ae44456050160347368b

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-01-10/1228/1

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-25-fortress-edifice-citadel/1343/1

**Pull request merged**